doc/source/train/user-guides/checkpoints.rst

```diff
@@ -174,6 +174,127 @@ appropriately in distributed training.
                         # Report to train session
                         checkpoint = Checkpoint.from_directory(tmpdir)
                         ray.train.report(metrics=metrics, checkpoint=checkpoint)
+
+
+    .. tab-item:: Hugging Face Transformers
+
+        Ray Train leverages HuggingFace Transformers Trainer's Callback 
+        to report metrics and checkpoints. 
+        
+        **Option 1: Use Ray Train's default report callback**
+        
+        We provide a simple callback implementation :class:`~ray.train.huggingface.transformers.RayTrainReportCallback` that 
+        reports on checkpoint save. It collects the latest logged metrics and report them together with the 
+        latest saved checkpoint.
+
+        .. code-block:: python
+            :emphasize-lines: 51,52
+
+            from ray.train.huggingface.transformers import (
+                RayTrainReportCallback,
+                prepare_trainer
+            )
+            from ray.train.torch import TorchTrainer
+            from transformers import TrainingArguments
+
+            def train_func(config):
+                ...
+
+                # Properly configure the `logging_strategy`, `save_strategy`, and 
+                # `evaluation_strategy` to ensure that the monitoring metrics
+                # are always logged at the same step of checkpoint saving.
+
+                # Valid configurations for eval metrics
+                args = TrainingArguments(
+                    ...,
+                    evaluation_strategy="epoch",
+                    save_strategy="epoch",
+                )
+
+                args = TrainingArguments(
+                    ...,
+                    evaluation_strategy="steps",
+                    save_strategy="steps",
+                    eval_steps=100,
+                    save_steps=100,
+                )
+
+                args = TrainingArguments(
+                    ...,
+                    evaluation_strategy="steps",
+                    save_strategy="steps",
+                    eval_steps=50,
+                    save_steps=100,
+                )
+
+                # Invalid configuration for eval metrics
+                invalid_args = TrainingArguments(
+                    ...,
+                    evaluation_strategy="steps",
+                    save_strategy="steps",
+                    save_steps=100,
+                    eval_steps=70,
+                )
```

Comment:
Oh I would exclude these from this code block. In this code snippet we can leave the first one in here, and have a comment above with something like:

```
# Configure logging_strategy  and saving_strategy as you normally would
```

The idea is that the focus of this block is to show the user that our callback is complementary with Transformers, and then a second section could  be for tips on ensuring compatibility.

Reply:
oh i see. That makes sense. We don't want to restrict the usability of the original transformers trainer.

Updated!